### PR TITLE
Modifying Admin v1 REST APIs to eliminate the limitations stated in wso2/product-apim-tooling#335

### PR DESCRIPTION
**Pull request by wasuradananjith**
## Purpose
Fixes: https://github.com/wso2/product-apim-tooling/issues/335

## Goals
Modify Admin v1 REST APIs to perform the tasks risen from https://github.com/wso2/product-apim-tooling/issues/335

## Approach
- Updated **Admin v1 GET /applications:** resource by adding a new optional parameter as **name**.
- Added a new resource as **Admin v1 DELETE /applications/{applicationId}:** which allows to delete an application for the user's who has **apim:admin** or **apim:app_import_export** scope.

For further clarification, refer [Mail: [Architecture] [APIM] Improving the functionality to delete applications using the API Controller by considering the -o (--owner) flag](http://wso2-oxygen-tank.10903.n7.nabble.com/APIM-Improving-the-functionality-to-delete-applications-using-the-API-Controller-by-considering-the-g-td161806.html)

## User stories
- Below cURL requests can be sent to retrieve applications using the modified  **Admin v1 GET /applications:** resource.

  **Example 1**
  ```
  curl -X GET "https://localhost:9443/api/am/admin/v1/applications?user=subscriber&name=testapp" -H "accept: application/json" -H "Authorization: Bearer 434c0df4-0943-3aa1-a55b-55e04a112bdc" -k
  ```
  This will retrieve the "testapp" which belongs to "subscriber".

  **Example 2**
  ```
  curl -X GET "https://localhost:9443/api/am/admin/v1/applications?name=testapp" -H "accept: application/json" -H "Authorization: Bearer 434c0df4-0943-3aa1-a55b-55e04a112bdc" -k
  ```
  This will retrieve the "testapp" which belongs to the currently logged in user.

- Below cURL request can be sent to delete an application specified by the applicationId which belongs to the same user or belongs to another user, using the newly added  **Admin v1 DELETE /applications/{applicationId}:** resource.

  **Example**
  ```
  curl -X DELETE "https://localhost:9443/api/am/admin/v1/applications/edc4f9ee-93fb-4863-bb8d-9de956e6ea9c" -H "accept: application/json" -H "Authorization: Bearer 434c0df4-0943-3aa1-a55b-55e04a112bdc" -k
  ```
  This will delete the application specified by "edc4f9ee-93fb-4863-bb8d-9de956e6ea9c".

## Documentation
Documentation changes should be done especially in the API Controller related docs.

## Related PRs
https://github.com/wso2/product-apim-tooling/pull/394

## Test environment
- JDK 1.8.0_251
- Ubuntu 20.04 LTS

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml [hunk 1]
```diff
@@ -1893,7 +1899,61 @@ paths:
...
+    delete:
+      security:
+        - OAuth2Security:
+            - apim:admin
+            - apim:app_import_export
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" -X DELETE \"https://localhost:9443/api/am/admin/v1/applications/367a2361-8db5-4140-8133-c6c8dc7fa0c4\""
+      x-wso2-request: |
+        DELETE https://localhost:9443/api/am/admin/v1/applications/367a2361-8db5-4140-8133-c6c8dc7fa0c4
+        Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
+      x-wso2-response: "HTTP/1.1 200 OK"
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml:1919]: shall we use x-examples format?

Comment (wasuradananjith) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml]: Sure

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/docs/examples/applications/applications.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      method: DELETE
+      url: https://localhost:9443/api/am/admin/v1/applications/367a2361-8db5-4140-8133-c6c8dc7fa0c4
+      headers: |
+        Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
+    response:
+      status:
+        code: 200
+        msg: OK
+      headers: |
+        Content-Type: application/json
```
Comment (Meruja) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/docs/examples/applications/applications.yaml:13]: Don't we need a new line eof?

Comment (wasuradananjith) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/docs/examples/applications/applications.yaml:13]: Fixed via https://github.com/wso2/carbon-apimgt/pull/9604/commits/65c3eb3ee90be6b18f554501b649a7bcfa4e3473

Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/docs/examples/applications/applications.yaml:13]: Thanks

**Pull request merged**